PyInstaller/building/makespec.py

```diff
@@ -263,6 +268,15 @@ def main(scripts, name=None, onefile=None,
         # The the text value 'None' means - use default icon.
         icon_file = 'None'
 
+    converted_datas = []
+    if type(datas) == list:
+        for all in datas:
+            pair = []
+            for entry in all.split(";"):
+                pair.append(entry)
+            converted_datas.append(tuple(pair))
+        datas = converted_datas
```

Comment:
This whole conversion could be done more elegant by passing a call-back to `type` in `g.add_argument('--datas',` (see .example in the [argparse docs](https://docs.python.org/2/library/argparse.html#type). This would allow raising a suitable error in case of wrong input-values.

(Beside of this:
- This code would allow passing *multiple* semi-colons (e.g. `foo;;;;bar;;`), which it must not.
- This is a typical case for a list-comprehension (unteted): `datas = [tuple(e.split(';') for e in datas]`
- The last point shows, that it's much better to do this in a call-back, since otherwise, errorhandling would be required here, too.)